go.mod

```diff
@@ -0,0 +1,83 @@
+module sigs.k8s.io/secrets-store-sync-controller
+
+go 1.22
+
+require (
+	github.com/google/go-cmp v0.6.0
+	github.com/prometheus/client_golang v1.16.0
+	go.opentelemetry.io/otel/exporters/prometheus v0.38.1
+	go.opentelemetry.io/otel/metric v0.38.1
+	go.opentelemetry.io/otel/sdk/metric v0.38.1
+	golang.org/x/crypto v0.17.0
+	google.golang.org/grpc v1.59.0
+	google.golang.org/protobuf v1.31.0
+	k8s.io/api v0.28.3
+	k8s.io/apimachinery v0.28.3
+	k8s.io/client-go v0.28.3
+	k8s.io/klog/v2 v2.100.1
+	k8s.io/utils v0.0.0-20240310230437-4693a0247e57
+	sigs.k8s.io/controller-runtime v0.16.3
+	sigs.k8s.io/secrets-store-csi-driver v1.4.1
```

Comment:
The fake provider in driver `v1.4.4` doesn't check for target path, so you can bump to that and remove the fake_server from here. This can be done in a follow-up.

Reply:
https://github.com/kubernetes-sigs/secrets-store-sync-controller/issues/18